### PR TITLE
expose minio's BucketLookupType to tempo's S3 config

### LINKED ISSUE #2109: Allow to configure RequestPathStyle within S3
**Is your feature request related to a problem? Please describe.**
Tempo's S3 config does not support set minio's BucketLookupType，for examples: The Volcano Engine product TOS S3 protocol request style only supports virtual host（https://www.volcengine.com/docs/6349/147050）

**Describe the solution you'd like**
expose the configuration

### PR DESCRIPTION
**Pull request by yeabow**
Minio Sdk supports it，So we need to expose the configuration. For examples: The Volcano Engine product TOS S3 protocol request style only supports virtual host（https://www.volcengine.com/docs/6349/147050）

**What this PR does**:

**Which issue(s) this PR fixes**:
Fixes #2109

**Checklist**
- [x] Tests updated
- [ ] Documentation added
- [x] [ENHANCEMENT] expose minio's BucketLookupType to tempo's S3 config

### CODE REVIEW

File: tempodb/backend/s3/s3.go [hunk 1]
```diff
@@ -386,6 +386,8 @@ func createCore(cfg *Config, hedge bool) (*minio.Core, error) {
 
 	if cfg.ForcePathStyle {
 		opts.BucketLookup = minio.BucketLookupPath
+	} else {
```
Comment (zalegrala) [inline tempodb/backend/s3/s3.go:389]: If both options can be false, then perhaps we want an `else if` here.  Are these options mutually exclusive?

**Review by zalegrala — COMMENTED** [review]
Thanks for the PR, what does BucketLookupType do?

File: tempodb/backend/s3/s3.go (hunk 1 above)
Comment (yeabow) [reply tempodb/backend/s3/s3.go:389]: minio.BucketLookup default value is  Auto(Zero)，RequestPathStyle is optional：0,1,2.

Comment (yeabow) [reply tempodb/backend/s3/s3.go:389]: ForcePathStyle Highest priority.

**Comment by yeabow** [comment]
> Thanks for the PR, what does BucketLookupType do?

BucketLookupAuto

**Review by joe-elliott — COMMENTED** [review]
I'm good on this but can we add a line to the docs:

https://github.com/grafana/tempo/blob/main/docs/sources/configuration/_index.md?plain=1#L660

**Comment by yeabow** [comment]
> I'm good on this but can we add a line to the docs:
> 
> https://github.com/grafana/tempo/blob/main/docs/sources/configuration/_index.md?plain=1#L660

done.

**Comment by zalegrala** [comment]
This looks good to me, thanks for putting this together.  Could you rebase for the conflict?

**Comment by joe-elliott** [comment]
I thought this PR was a nice addition. Any particular reason it was closed?

**Comment by yeabow** [comment]
> I thought this PR was a nice addition. Any particular reason it was closed?

https://github.com/grafana/tempo/pull/2123

**Comment by zalegrala** [comment]
I think what happened is the `main` commit was force pushed to this branch, so there was no diff, and github closed it.  Sorry for the trouble.  Will review the other PR you've opened.

**Pull request closed without merging**